signature/policy_eval_signedby.go

```diff
@@ -70,6 +71,13 @@ func (pr *prSignedBy) isSignatureAuthorAccepted(image types.UnparsedImage, sig [
 			return PolicyRequirementError(fmt.Sprintf("Signature by key %s is not accepted", keyIdentity))
 		},
 		validateSignedDockerReference: func(ref string) error {
+			r := image.Reference().DockerReference()
+			if _, isCanonical := r.(reference.Canonical); isCanonical || reference.IsNameOnly(r) {
+				if !pr.UntaggedReference.matchesDockerReference(image, ref) {
+					return PolicyRequirementError(fmt.Sprintf("Signature for identity %s is not accepted", ref))
+				}
+				return nil
```

Comment:
right, not sure I fully understand what you want to see though, `matchesDockerReference(image types.UnparsedImage, signatureDockerReference string, untaggedPolicy PolicyUntaggedReferenceMatch)`? where `PolicyUntaggedReferenceMatch`:
```
type PolicyUntaggedReferenceMatch interface {
    // ???
}
```